x/nexus/keeper/grpc_query.go

```diff
@@ -233,3 +233,41 @@ func (q Querier) RecipientAddress(c context.Context, req *types.RecipientAddress
 		RecipientChain: linkedAddresses.RecipientAddress.Chain.Name.String(),
 	}, nil
 }
+
+// TransferRateLimit queries the transfer rate limit for a given chain and asset
+func (q Querier) TransferRateLimit(c context.Context, req *types.TransferRateLimitRequest) (*types.TransferRateLimitResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+
+	chain, ok := q.keeper.GetChain(ctx, nexus.ChainName(req.Chain))
+	if !ok {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrNotFound, fmt.Errorf("chain %s not found", req.Chain).Error())
+	}
+
+	rateLimit, found := q.keeper.getRateLimit(ctx, chain.Name, req.Asset)
+	if !found {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrNotFound, fmt.Errorf("rate limit not found for chain %s and asset %s", chain.Name, req.Asset).Error())
```

Comment:
should return unlimited instead of error to reflect the actual behaviour

Reply:
honestly, with a default fallback, it doesn't really make sense to me to have the `found` return value at all. I'd prefer to have `getRateLimit` return a value in all cases, then the calling code doesn't have to make additional decisions